core/Policy/CnilPolicy.php

```diff
@@ -20,7 +20,21 @@ public static function getName(): string
 
     public static function getDescription(): string
     {
-        return Piwik::translate('General_ComplianceCNILDescription');
+        $description = Piwik::translate('General_ComplianceCNILDescription');
+
+        $isCloud = false;
+
+        /**
+         * This event should only be used by the cloud plugin, to determine that the
+         * current instance is a cloud instance.
+         */
+        Piwik::postEvent('Policy.onCloudInstance', [$isCloud]);
+
+        if ($isCloud) {
+            $description .= ' ' . Piwik::translate('General_ComplianceDPALink', ['<a href="https://matomo.org/matomo-cloud-dpa/">', '</a>']);
+        }
+
+        return $description;
```

Comment:
As we don't want any cloud specific code in core, we should only post an event to update the description, and handle the rest in the cloud plugin.

Passing the policy class through the event as well if the event consumer wanted different updates per policy.

```suggestion
        $description = Piwik::translate('General_ComplianceCNILDescription');
        
        Piwik::postEvent('CompliancePolicy.updateDescription', [&$description, self::class]);

        return $description;
```